plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/state/FollowPuckViewportStateImpl.kt

```diff
@@ -103,6 +113,8 @@ internal class FollowPuckViewportStateImpl(
       locationComponent.removeOnIndicatorPositionChangedListener(indicatorPositionChangedListener)
       locationComponent.removeOnIndicatorBearingChangedListener(indicatorBearingChangedListener)
       isObservingLocationUpdates = false
+      lastBearing = null
+      lastLocation = null
```

Comment:
Wonder why is last location and bearing reset here?

Reply:
The way I understand it is because we stop listening for updates therefore we can't guarantee that they're the most up-to-date values after a while.

Reply:
I think it's worth adding a comment to the code explaining it. (whatever the reason is)

Reply:
Added some inline comments.